scrapers_next/va/committees.py

```diff
@@ -0,0 +1,149 @@
+            # .do_scrape() allows us to get information from MemberDetail without
+            # returning/writing a com object to disk
+            com = [
+                i
+                for i in MemberDetail(
+                    [com, role], source=URL(detail_link, timeout=120)
+                ).do_scrape()
+            ][0]
+
+        return com
+
+
+class FindSubCommittees(HtmlListPage):
+
+    selector = CSS("#mainC > ul:nth-child(12) > li a")
+
+    def process_item(self, item):
+        try:
+            comm_name = item.text
+
+        except SelectorError:
+            raise SkipItem("no subcommittees")
+
+        chamber_text = item.getparent().getparent().getparent().getchildren()[1].text
+        if "house" in chamber_text.lower():
+            chamber = "lower"
+            parent_comm = chamber_text.split("House")[1].strip()
+        else:
+            chamber = "upper"
+            parent_comm = chamber_text.split("Senate")[1].strip()
+
+        com = ScrapeCommittee(
+            name=comm_name,
+            classification="subcommittee",
+            chamber=chamber,
+            parent=parent_comm,
+        )
+
+        detail_link = item.get("href")
+        com.add_source(self.source.url, note="homepage")
+        com.add_source(detail_link)
+
+        return CommitteeDetail(com, source=URL(detail_link, timeout=120))
+
+
+class CommitteeList(HtmlListPage):
+    source = "https://lis.virginia.gov/231/com/COM.HTM"
+    selector = CSS(".linkSect a")
+
+    def process_item(self, item):
+        comm_name = item.text
+        print(comm_name)
+        # both senate and house committees are listed on one page, so this isolates which is which
+        chamber_text = item.getparent().getparent().getparent().getchildren()[0].text
+        if "HOUSE" in chamber_text:
+            chamber = "lower"
+        else:
+            chamber = "upper"
+
+        com = ScrapeCommittee(
+            name=comm_name,
+            classification="committee",
+            chamber=chamber,
+        )
+
+        detail_link = item.get("href")
+        com.add_source(self.source.url, note="homepage")
+        com.add_source(detail_link)
+        time.sleep(3)
+
+        return CommitteeDetail(com, source=URL(detail_link, timeout=120))
+
+
+class SubCommitteeList(HtmlListPage):
+    source = "https://lis.virginia.gov/231/com/COM.HTM"
+    selector = CSS(".linkSect a")
+
+    def process_item(self, item):
+        chamber_text = item.getparent().getparent().getparent().getchildren()[0].text
+        comm_name = item.text
+
+        # this is hacky but I couldn't figure out another way to ignore a lack of subcommittees
+        if "SENATE" in chamber_text:
+            if comm_name == "Transportation":
+                raise SkipItem("no subcommittees")
+            if comm_name == "Rules":
+                time.sleep(4)
+                raise SkipItem("no subcommittees")
+
+        detail_link = item.get("href")
+        time.sleep(4)
```

Comment:
Again, we should either clarify _why_ we're sleeping or remove the `sleep` statement.

Reply:
ok! sleep statements helped the scraper run more smoothly (was getting a lot of SelectorErrors or source errors which I believe comes from the VA site being a little slow). I can add that in as a comment, unless you think I should be doing something else!

Reply:
I think you want `SCRAPELIB_RPM`:
```
scrapers/ca/__init__.py:settings = dict(SCRAPELIB_RPM=30)
scrapers/ct/__init__.py:settings = {"SCRAPELIB_RPM": 20}
scrapers/ia/__init__.py:settings = {"SCRAPELIB_RPM": 30}
scrapers/ks/__init__.py:settings = dict(SCRAPELIB_TIMEOUT=300, SCRAPELIB_RPM=12)
scrapers/nd/__init__.py:    "SCRAPELIB_RPM": 10,
scrapers/pa/__init__.py:settings = {"SCRAPELIB_RPM": 30}
scrapers/va/__init__.py:settings = {"SCRAPELIB_RPM": 40}
scrapers/vt/__init__.py:settings = dict(SCRAPELIB_RPM=20)
```
It's built-in rate-limiting that slows down requests to remote sites. Although there may be a similar function in spatula?

Reply:
The point is that having random sleeps in a code base makes debugging inconsistent and can unintentional suppress (or create) problems that are difficult to find.

Reply:
Gotcha. spatula has a way to set requests per minute from the command line. i'll take out the sleep statements and use that instead